opentracing/src/main/java/opentracing/TraceContextSource.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing;
+
+import java.util.Map;
+
+/**
+ * Long-lived interface that knows how to create a root {@linkplain TraceContext} and encode/decode
+ * any other.
+ */
+public interface TraceContextSource {
```

Comment:
Sure, yeah... if you are writing the code that serializes a `TraceContext` to the wire (e.g., in HTTP headers, or stuffed into a binary protocol like thrift or grpc somewhere) – or deserializes the `TraceContext` on the other side – the `codec()` is needed. The non-`codec()` parts of the `TraceContextSource` are useful for code that can be shared by most `Tracer` implementations... less important than the `codec()` in my mind, but important nonetheless. If you speak Go, this is how the "standard" `Tracer` uses `TraceContextSource`: https://github.com/opentracing/opentracing-go/blob/master/standardtracer/standardimpl.go#L129 (there are other similar call sites in that file).